src/consent-string.js

```diff
@@ -0,0 +1,343 @@
+const { encodeConsentString } = require('./encode');
+const { decodeConsentString } = require('./decode');
+
+/**
+ * Regular expression for validating
+ */
+const consentLanguageRegexp = /^[a-z]{2}$/;
+
+class ConsentString {
+  /**
+   * Initialize a new ConsentString object
+   *
+   * @param {string} baseString An existing consent string to parse and use for our initial values
+   */
+  constructor(baseString = null) {
+    this.created = new Date();
```

Comment:
Should these only be created if `baseString === null?`   You could move this down to the `else` of the if baseString